src/mmds/src/lib.rs

```diff
@@ -138,16 +195,70 @@ fn respond_to_get_request(request: Request) -> Response {
         }
     };
 
+    // Allow only GET and PUT requests.
+    match request.method() {
+        Method::Get => respond_to_get_request_checked(request, token_headers),
+        Method::Put => respond_to_put_request(request, token_headers),
+        _ => {
+            let mut response = build_response(
+                request.http_version(),
+                StatusCode::MethodNotAllowed,
+                Body::new(Error::MethodNotAllowed.to_string()),
+            );
+            response.allow_method(Method::Get);
+            response.allow_method(Method::Put);
+            response
+        }
+    }
+}
+
+fn respond_to_get_request_checked(request: Request, token_headers: TokenHeaders) -> Response {
+    // Get MMDS token from custom headers.
+    let token = match token_headers.x_metadata_token() {
+        Some(token) => token,
+        None => {
+            let error_msg = Error::NoTokenProvided.to_string();
+            return build_response(
+                request.http_version(),
+                StatusCode::Unauthorized,
+                Body::new(error_msg),
+            );
+        }
+    };
+
+    // Validate MMDS token.
+    let is_valid = match MMDS.lock().expect("Poisoned lock").as_ref() {
+        Some(mmds) => mmds.is_valid_token(token),
```

Comment:
We can have the check I mentioned in the other comment (regarding `unreachable!()`) here, especially since this is an internal function and we can better control the state machine of the MMDS object from here to actually ensure we never call `is_valid_token` in a context in which it would panic.